profiler/src/ProfilerEngine/Datadog.Profiler.Native/CorProfilerCallback.cpp

```diff
@@ -341,6 +349,10 @@ bool CorProfilerCallback::InitializeServices()
         _pAllocationsRecorder.get()
         );
 
+    _processSamplesProvider = std::make_unique<ProcessSamplesProvider>(_garbageCollectorInfo.get(), _pCpuTimeProvider);
```

Comment:
But what if cpu profiling is disabled by configuration?
Do we need to create a "fake" one?

Reply:
rewritten :)